typescript: CustomPatternMatcherFunc return type should be "RegExpExecArray | null"

Hi, I would like to switch to Typescript in my project. I use a lot of CustomPatternMatcherFuncs in my lexer which forces me to return RegExpExecArray even if the match returns null. So I tried to return `<RegExpExecArray>{}` if nothing was matched, but this leads to an error in https://github.com/SAP/chevrotain/blob/master/src/scan/lexer_public.ts#L521 because imageLength is not null and empty. 

Could you please change the return type to `RegExpExecArray | null`?

I will give make some changes, but I probably do not know enough about all the functions to make the decision in all cases. 

By the way: I ran the dev_integration_tests in a new clone and a JSON test failed (I wanted to make sure that I am not responsible): 

````
  1) The JSON Grammar with comments
       can parse a simple Json without errors:
     AssertionError: expected [ Array(1) ] to be empty
      at Context.it (json/json_with_comments_spec.js:22:46)
```

--- a/api.d.ts
+++ b/api.d.ts
@@ -1313,7 +1313,7 @@ export declare type CustomPatternMatcherFunc = (
     groups?: {
         [groupName: string]: IToken
     }
-) => RegExpExecArray
+) => RegExpExecArray | null
 
 /**
  *  API #2 for [Custom Token Patterns](http://sap.github.io/chevrotain/docs/guide/custom_token_patterns.html).

